internal/rpc/clients.go

```diff
@@ -71,14 +73,14 @@ func GRPCClientFromConfig(cfg config.View, prefix string) (*grpc.ClientConn, err
 	}
 
 	// If TLS support is enabled in the config, fill in the trusted certificates for decrpting server certificate.
-	if cfg.GetBool("tls.enabled") {
-		_, err := os.Stat(cfg.GetString("tls.trustedCertificatePath"))
+	if cfg.GetString(configNameClientTrustedCertificatePath) != "" {
```

Comment:
cfg.IsSet(configNameClientTrustedCertificatePath)

Reply:
IsSet won't work. That's if the key doesn't exist. If it's "" it'll try to do TLS mode which will be bad.